### ISSUE TITLE
Missing start() method on EventEmitter

### ISSUE THREAD

**Issue by jlundan** [FlorianKoerner/nodegit-typescript-definition#1]
First of all, thanks for the typings! I am working on ts project which uses nodegit and found your definitions file. =)

Found an issue though (I think) related to getting commit history from master commit. Nodegit says in their example, to get commit history:

```javascript
var Git = require("nodegit");

// Open the repository directory.
Git.Repository.open("tmp")
  // Open the master branch.
  .then(function(repo) {
    return repo.getMasterCommit();
  })
  // Display information about commits on master.
  .then(function(firstCommitOnMaster) {
    // Create a new history event emitter.
    var history = firstCommitOnMaster.history();

    // Create a counter to only show up to 9 entries.
    var count = 0;

    // Listen for commit events from the history.
    history.on("commit", function(commit) {
      // Omitting this part from the example

    });

    // Start emitting events.
    history.start();
  });
```
I now have in typescript
```typescript
import {autoinject} from 'aurelia-framework';
/// <reference path="/custom_typings/nodegit.d.ts" />
import {Repository} from "nodegit";

@autoinject
export class GitService {
    constructor() { }

    public listCommits(path: string, limit: number): void {
        Repository.open(path).then(function (repo) {
           return repo.getMasterCommit();
        })
        .then(function (firstCommitOnMaster) {         
            var history = firstCommitOnMaster.history();
            var count = 0;

            history.on("commit", function (commit) {
                  // Omitting this part from the example
            });

            history.start();
        });
    }
}
```
The problem is that typescript compiler says that NodeJS.EventEmitter does not have an start method. If I look at the nodegit history method in nodegit's commit.js
```javascript
Commit.prototype.history = function () {
  var event = new events.EventEmitter();
  var oid = this.id();
  var revwalk = this.repo.createRevWalk();

  revwalk.sorting.apply(revwalk, arguments);

  var commits = [];

  event.start = function () {
    revwalk.walk(oid, function commitRevWalk(error, commit) {
      if (error) {
        if (error.errno === NodeGit.Error.CODE.ITEROVER) {
          event.emit("end", commits);
          return;
        } else {
          return event.emit("error", error);
        }
      }

      event.emit("commit", commit);
      commits.push(commit);
    });
  };

  return event;
};
```

I can see that it actually returns nodejs EventEmitter, but before that it dynamically adds the start method. Typings file on the other hand returns NodeJs.EventEmitter in the Commit class

```typescript
history(): NodeJS.EventEmitter;
```
Since the start method is dynamically added to the events.EventEmitter class by nodegit, NodeJS.EventEmitter interface does not have it. I have fixed this by implementing EventEmitter interface in the NodeGit namespace in the typings file:

```typescript
    export interface EventEmitter extends NodeJS.EventEmitter {
        start(): void;
    }
```

and made the history method return that.
```typescript
history(): EventEmitter;
```

That makes nodegit correctly read the commit list in my project. I can do pull request if you agree with this and wish to include this change.

**Comment by FlorianKoerner** [comment]
Thank you for your detailed description. Can you create a pull request for this change? 😃

**Comment by thyb** [comment]
I've opened a pull request with the change suggested by @jlundan

**Comment by xogeny** [comment]
Note that a similar issue (and fix) can be found in #4.
